meta-balena-common/recipes-core/initrdscripts/files/migrate

```diff
@@ -106,6 +94,11 @@ migrate_enabled() {
     return 1
 }
 
+fail_copy() {
+    relocate_log_mount
+    fail "Unable to copy files to memory"
+}
+
```

Comment:
Please test that the log files are available to debug this error condition.

Reply:
@alexgg What's the intended behavior here? I wouldn't expect a move mount to copy files to another filesystem, only move where the mount is accessible from. Maybe I'm missing something.

I tried retaining the existing failure handling mechanism.

Reply:
@alexgg I ran the image with the `shell` parameter and verified that `/run` is a tmpfs mount. I don't see how moving the mount to the rootfs would ever persist any files from this mount.

Reply:
@jakogut what I want to make sure is that when the migrate tasks fails because of lacking memory, on reboot the migrator log file is accessible and we can debug the problem and not just have a silent failure.

How I did this in the past is using the Windows migrator - start with a Windows 10 on a VM with not enough memory, run the migration, see it fail and then check that the `resin-boot` partition in the main drive has the migrate log file with a descriptive message.

It's probably fine as you keep the same logic, but unless it's verified on change we have no other way or realizing apart from a customer hitting the problem and realizing we don't even have a log file.

Reply:
Migrator releases are in https://github.com/balena-os/migrator/releases

Reply:
@alexgg This was more complicated and challenging than I expected, but I think I've come up with a good solution.

Previously, any failures prior to `resin-init-flasher` running wouldn't be logged, as the log file was copied from tmpfs to persistent storage in that script. I worked around this by redirecting stdout/stderr to tee, which then directly writes that output to a logfile in both tmpfs, as well as persistent storage. This has the added benefit of continuing to log to disk even when `shell_debug` is enabled.

This handles logging in the case where migration fails before `resin-init-flasher` is executed, but we still need to write a log file to tmpfs in that script, as we wipe out the boot partition filesystem during migration. After that partition is written, we can then copy the log to that filesystem.

Here's the output from such a failed attempt:
```
zramctl: /dev/zram1: failed to reset: Device or resource busy
[init][INFO] Created 2.25 GB zram device using lz4 alg
Filesystem label=
OS type: Linux
Block size=4096 (log=2)
Fragment size=4096 (log=2)
147456 inodes, 589824 blocks
29491 blocks (5%) reserved for the super user
First data block=0
Maximum filesystem blocks=4194304
18 block groups
32768 blocks per group, 32768 fragments per group
8192 inodes per group
Superblock backups stored on blocks:
	32768, 98304, 163840, 229376, 294912
[init][ERROR] Unable to copy files, not enough memory
```

Here's the log of a successful migration:
```
[init][INFO] Flasher detected. Avoiding filesystem UUIDS initialization.
[INFO] Flasher detected. We don't run fsck on the partitions when provisioning.
[init][INFO] Flasher detected. Avoiding expand partition mechanism.
[init][INFO] Using bootparam_root /dev/disk/by-label/flash-rootA for flash-rootA
[init][INFO] Using /dev/disk/by-label/flash-rootA as root filesystem.
[INFO] Flasher detected. Avoiding persistent machine-id.
[init][INFO] Searching for devices matching pattern /dev/md/balena{,_*}
[init][INFO] Searching for devices matching pattern /dev/nvme?n?
[init][INFO] Searching for devices matching pattern /dev/sd?
[init][INFO] Searching for devices matching pattern /dev/mmcblk?
[init][INFO] Searching for devices matching pattern /dev/vd?
[init][INFO] Migration requested in configuration
[init][INFO] Running migration on /dev/sda...
zramctl: /dev/zram1: failed to reset: Device or resource busy
[init][INFO] Created 2.25 GB zram device using lz4 alg
Filesystem label=
OS type: Linux
Block size=4096 (log=2)
Fragment size=4096 (log=2)
147456 inodes, 589824 blocks
29491 blocks (5%) reserved for the super user
First data block=0
Maximum filesystem blocks=4194304
18 block groups
32768 blocks per group, 32768 fragments per group
8192 inodes per group
Superblock backups stored on blocks:
	32768, 98304, 163840, 229376, 294912
/init.d/92-migrate: line 106: kill: (666) - No such process
/init.d/92-migrate: line 106: kill: (2759) - No such process
/init.d/92-migrate: line 106: kill: (2765) - No such process
/init.d/92-migrate: line 189: kill: (2827) - No such process
[resin-init-flasher][INFO] Board specific initialization...
[resin-init-flasher][INFO] resin-init-flasher configuration found.
[WARN] /usr/bin/resin-init-flasher: Missing fatrw command
[resin-init-flasher][INFO] balena configuration found.
[resin-init-flasher][INFO] Flash internal device... will take around 5 minutes... 
[resin-init-flasher][INFO] Searching for devices matching pattern /dev/md/balena{,_*}
[resin-init-flasher][INFO] Searching for devices matching pattern /dev/nvme?n?
[resin-init-flasher][INFO] Searching for devices matching pattern /dev/sd?
[resin-init-flasher][INFO] Searching for devices matching pattern /dev/mmcblk?
[resin-init-flasher][INFO] Searching for devices matching pattern /dev/vd?
[resin-init-flasher][INFO] /dev/sda will be used for flashing.
efivar: show variable: No such file or directory
efivar: show variable: No such file or directory
/usr/bin/resin-init-flasher: line 206: [: : integer expression expected
[resin-init-flasher][INFO] Unprovisioned: Percentage 0, status Starting flashing balenaOS on internal media
[resin-init-flasher][INFO] Unprovisioned: Percentage 100, status Flashing balenaOS on internal media
Warning: Not all of the space available to /dev/sda appears to be used, you can fix the GPT to use all of the space (an extra 12132352 blocks) or continue with the current setting? 
[resin-init-flasher][INFO] No need to flash first stage bootloader to a specific device.
[resin-init-flasher][INFO] No need to flash second stage bootloader to a specific device.
[resin-init-flasher][INFO] Mounting internal device boot partition.
[resin-init-flasher][INFO] Transferring system connections on the internal device.
'/mnt/boot/system-connections' -> '/tmp/internal_boot/system-connections'
'/mnt/boot/system-connections/balena-sample.ignore' -> '/tmp/internal_boot/system-connections/balena-sample.ignore'
'/mnt/boot/system-connections/README.ignore' -> '/tmp/internal_boot/system-connections/README.ignore'
[resin-init-flasher][INFO] Transferring proxy configuration on the internal device.
'/mnt/boot/system-proxy' -> '/tmp/internal_boot/system-proxy'
'/mnt/boot/system-proxy/redsocks.conf.ignore' -> '/tmp/internal_boot/system-proxy/redsocks.conf.ignore'
'/mnt/boot/system-proxy/README.ignore' -> '/tmp/internal_boot/system-proxy/README.ignore'
[resin-init-flasher][INFO] Board specific flash procedure...
[resin-init-flasher][INFO] Log end
```